### PR TITLE
Connect react and redux

### PR DESCRIPTION
**Pull request by Basir-Mohammadi**
## Proposed changes

The aim of this pull request is to add react and redux to the project

**The following setups were implemented**
- First of all - make sure that you keep state immutable in reducers!
- Secondly, change the default state in your books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.
- Import Redux Provider and your store in the main component of your React app.
- Connect your app to the Redux store.
- Make sure that list of books displays books from Redux store.
- Use useSelector to consume the state.
- In your React component responsible for adding new books, add functionality for "Add new" button click:
- Import Redux useDispatch.
- Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
- Takes the inputs from the state and generate a unique id and add them to an object.
- Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for removing books, add a functionality for "Remove" button click:
- Import Redux useDispatch.
- Dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for categories, add a functionality for "Check status" button click:
- Import Redux useDispatch.
- Dispatch a corresponding action (import Action Creator that you created before and use it here).

## Type of change

-   New feature

## Checklist

-    [Linting tests](https://github.com/microverseinc/linters-config) pass locally with my changes
-    Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md)
-    Documentation is done in a [professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,58 @@
...
+
+- Clone repository in your local machine 
+- cd Portfolio
+- open `index.html` in your browser.
+- You can use live server.
+
+## Live Demo
+
+[Books Live](https://basirking.github.io/Awesome-Book/)
+
```
Comment (zieeco) [inline README.md]: - Please, kindly remove or update the live demo link as it takes users to another project which is not a good user experience. please, kindly fix this.

File: README.md [hunk 2]
```diff
@@ -1 +1,58 @@
...
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
+- Inspiration
+- etc
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (zieeco) [inline README.md]: - Please, kindly ensure the MIT.md licensed content are available and not a broken link to make your readme more professional.

File: src/component/form.js [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
+  const title = useRef(null);
+  const author = useRef(null);
+  return (
+    <form className="forms" onSubmit={(e) => { e.preventDefault(); dispatch(addBook(title.current.value, author.current.value, uniqid())); e.target.reset(); }}>
+      <input ref={title} type="text" name="author" placeholder="Author" />
+      <input ref={author} type="text" name="book" placeholder="Book" />
+      <button type="submit">Add</button>
+    </form>
+  );
+};
```
Comment (zieeco) [inline src/component/form.js]: - Hey, please kindly ensure your form doesn't get submitted with an empty field. Please, kindly fix this issue. You can move the logic here in another function, and do some check to ensure the inputs fields are filled before the form get submitted

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Basir-Mohammadi,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @Basir-Mohammadi  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**